ts/a11y/speech/GeneratorPool.ts

```diff
@@ -0,0 +1,353 @@
+   */
+  public updateRegions(
+    node: N,
+    speechRegion: LiveRegion,
+    brailleRegion: LiveRegion
+  ) {
+    let speech = this.getLabel(node, this.lastSpeech);
+    speechRegion.Update(speech);
+    // TODO: See if we can reuse the speech from the speech region.
+    this.adaptor.setAttribute(node, 'aria-label', buildSpeech(speech)[0]);
+    if (this.lastSpeech) {
+      this.lastSummary = true;
+    }
+    this.lastSpeech = '';
+    brailleRegion.Update(
+      this.adaptor.getAttribute(node, 'aria-braillelabel'));
+  }
+
+  /**
+   * Updates the speech in the give node.
+   *
+   * @param {N} node The typeset node.
+   */
+  public updateSpeech(node: N) {
+    const xml = this.prepareXml(node);
+    const speech = this.speechGenerator.getSpeech(xml, this.element);
+    this.setAria(node, xml, this.options.sre.locale);
+    const label = buildSpeech(speech)[0];
+    this.adaptor.setAttribute(node, 'aria-label', label);
+    return label;
+  }
+
+  /**
+   * Cycles rule sets for the speech generator.
+   *
+   * @param {N} _node The typeset node.
+   */
+  public nextRules(_node: N) {
+    this.speechGenerator.nextRules();
+    this.updateSummaryGenerator();
+  }
+
+  /**
+   * Cycles style or preference settings for the speech generator.
+   *
+   * @param {N} node The typeset node.
+   */
+  public nextStyle(node: N) {
+    this.speechGenerator.nextStyle(
+      this.adaptor.getAttribute(node, 'data-semantic-id'));
+    this.updateSummaryGenerator();
+  }
+
+  /**
+   * Copies domain and style option from speech to summary generator. This is
+   * necessary after when either option is changed on the fly.
+   */
+  private updateSummaryGenerator() {
+    const options = this.speechGenerator.getOptions();
+    this.summaryGenerator.setOption('domain', options['domain']);
+    this.summaryGenerator.setOption('style', options['style']);
+  }
+
+  /**
+   * Makes a node amenable for SRE computations by reparsing.
+   *
+   * @param {N} node The node.
+   */
+  private prepareXml(node: N) {
+    return Sre.parseDOM(this.adaptor.serializeXML(node));
+  }
+
+  /**
+   * Speech, labels and aria
+   */
+
+  /**
+   * Computes the speech label from the node combining prefixes and postfixes.
+   *  
+   * @param {N} node The typeset node.
+   * @param {string=} center Core speech. Defaults to `data-semantic-speech`.
+   * @param {string=} sep The speech separator. Defaults to space.
+   */
+  public getLabel(node: N,
+                  center: string = '',
+                  sep: string = ' ') {
+    return buildLabel(
+      center || this.adaptor.getAttribute(node, 'data-semantic-speech'),
+      this.adaptor.getAttribute(node, 'data-semantic-prefix'),
+      // TODO: check if we need this or if it is automatic by the screen readers.
+      this.adaptor.getAttribute(node, 'data-semantic-postfix'),
+      sep
+    );
+  }
+
+  private copyAttributes(xml: Element, node: N, attr: string) {
```

Comment:
Do you need jsdoc for this function and the `attrList` below?

Reply:
I thought I had caught them all. 
jsdoc added.